src/cpp/builder/numeric.cc

```diff
@@ -5,6 +5,10 @@
 #include "builder/common.h"
 #include "builder/numeric.h"
 
+#ifdef _MSC_VER
```

Comment:
I normally prefer to account for platform-related logic into the build system for more flexibility instead of preprocessor. However, does this work when compiling using `mingw64` instead?

Reply:
Mingwin uses gcc no ? So I guess that's ok. I dont' like the if in the cmake file because it makes browsing the code a bit difficult

Reply:
Just FYI, tested locally with `mingw64` and it works. (I had to recompile `libpq` and `arrow-cpp` manually because C++ libraries compiled with MSVC cannot be linked against other binaries compiled with `mingw64` (different mangling standards))